File: tests/fixtures/check_then_act/ts/nested_callback_negative_fixture.ts
// Awaits inside callbacks nobody awaits are not gaps for the function that schedules them.
export async function importAll(repo: any, items: string[], key: string) {
  const existing = repo.find(key);
  if (existing) return existing;
  items.forEach(async (item) => {
    await repo.warm(item);
  });
  setTimeout(async () => {
    await repo.flush();
  }, 10);
  repo.insert(key);
}

File: tests/fixtures/check_then_act/py/nested_callback_negative_fixture.py
# Work handed to a task nobody awaits, or to a lambda, is not a gap for the function that schedules it.
import asyncio


async def import_all(repo, items, key):
    existing = repo.find(key)
    if existing:
        return existing
    for item in items:
        asyncio.create_task(repo.warm(item))
    asyncio.get_running_loop().call_later(10, lambda: repo.flush())
    repo.insert(key)
